### PR TITLE
fix codestylesettings

### PR DESCRIPTION
**Pull request by serious6**
Ref. to #158 Errors in project formatter [codeStyleSettings.xml]:
* comments should be formatted now
* added some other missing parameters

feel free to help verifying this fix [here](https://code.google.com/p/google-styleguide/source/browse/trunk/intellij-java-google-style.xml)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@serious6__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by serious6** [comment]
_this is a replacement of #212_

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -3,20 +3,65 @@
...
+          <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
+          <option name="PARENT_SETTINGS_INSTALLED" value="true" />
+        </codeStyleSettings>
         <codeStyleSettings language="JAVA">
           <option name="KEEP_FIRST_COLUMN_COMMENT" value="false" />
           <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
           <option name="KEEP_BLANK_LINES_IN_DECLARATIONS" value="3" />
-          <option name="KEEP_BLANK_LINES_IN_CODE" value="3" />
+          <option name="KEEP_BLANK_LINES_IN_CODE" value="1" />
           <option name="KEEP_BLANK_LINES_BEFORE_RBRACE" value="3" />
```
Comment (vboctor) [inline .idea/codeStyleSettings.xml]: Why do we allow blank lines before right brace?  Is this part of the standard?

**Comment by vboctor** [comment]
Looks good.  So after this gets in, are you going to have a checkin for fixing the formatting of existing code?

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (serious6) [inline .idea/codeStyleSettings.xml]: fixed

**Comment by serious6** [comment]
@vboctor I can do that yes. But perhabs with a seperated PR. Currently I am also working on #39.

File: .idea/codeStyleSettings.xml [hunk 2]
```diff
@@ -3,20 +3,64 @@
...
+        <option name="JD_P_AT_EMPTY_LINES" value="false" />
+        <option name="JD_KEEP_EMPTY_PARAMETER" value="false" />
+        <option name="JD_KEEP_EMPTY_EXCEPTION" value="false" />
+        <option name="JD_KEEP_EMPTY_RETURN" value="false" />
+        <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
+        <option name="WRAP_COMMENTS" value="true" />
         <XML>
           <option name="XML_LEGACY_SETTINGS_IMPORTED" value="true" />
         </XML>
+        <codeStyleSettings language="Groovy">
```
Comment (evpaassen) [inline .idea/codeStyleSettings.xml]: I don't think this project needs Groovy code style settings. Or does it?

File: .idea/codeStyleSettings.xml [hunk 3]
```diff
@@ -3,20 +3,64 @@
...
+          <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
+          <option name="PARENT_SETTINGS_INSTALLED" value="true" />
+        </codeStyleSettings>
         <codeStyleSettings language="JAVA">
           <option name="KEEP_FIRST_COLUMN_COMMENT" value="false" />
           <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
           <option name="KEEP_BLANK_LINES_IN_DECLARATIONS" value="3" />
-          <option name="KEEP_BLANK_LINES_IN_CODE" value="3" />
-          <option name="KEEP_BLANK_LINES_BEFORE_RBRACE" value="3" />
```
Comment (evpaassen) [inline .idea/codeStyleSettings.xml]: Where did this setting go?

Comment (evpaassen) [inline .idea/codeStyleSettings.xml]: Looks like it's 2 by default. The Google code style says:

> ##### 4.6.1 Vertical Whitespace
> ...
> Multiple consecutive blank lines are permitted, but never required (or encouraged).

I'd say we'd set all `KEEP_BLANK_LINES_IN_*` to `2`.

**Comment by evpaassen** [comment]
I created a patch with some enhancements which you can apply to your branch:
https://gist.github.com/evpaassen/df338eb728cc1d9d2ea7

It addresses the following issues:
* Keep line breaks when reformatting should be false (`KEEP_LINE_BREAKS`). The codebase currently has many line breaks on the most random places.
* <p/> should be inserted between paragraphs in Javadoc (`JD_P_AT_EMPTY_LINES`).
* Max. 2 consecutive blank lines (`KEEP_BLANK_LINES_IN_*`).
* Blank line between license header and package declaration (`BLANK_LINES_BEFORE_PACKAGE`).
* Order of imports, grouping and blank lines between groups, like the Google code style [requires](https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s3.3-import-statements).  
* Removed Groovy stuff.

Apart from that, your config looks fine, as far as I can tell.

**Comment by serious6** [comment]
@evpaassen thanks for the patch.

**Comment by serious6** [comment]
` <option name="KEEP_BLANK_LINES_IN_CODE" value="1" />` has now been removed. But if you have a look at [intellij-java-google-style.xml#260](https://code.google.com/p/google-styleguide/source/browse/trunk/intellij-java-google-style.xml#260) this one is set. Do we want to keep this?

**Comment by evpaassen** [comment]
> `<option name="KEEP_BLANK_LINES_IN_CODE" value="1" />` has now been removed. But if you have a look at intellij-java-google-style.xml#260 this one is set. Do we want to keep this?

Missing means `2` (IntelliJ default), which I think is valid according to the Google guide.

**Pull request merged**